### PR TITLE
[FIX] upgrade packages to latest and upgrade golang to 1.21

### PR DESCRIPTION
**Pull request by hackerghost93**

### CODE REVIEW

File: watcher_test.go [hunk 1]
```diff
@@ -82,21 +82,26 @@ func testCancelByFunc(t *testing.T, fn func(wg *sync.WaitGroup, loader *Loader,
 	assert.Equal(t, 0, countWatchers(loader))
 }
 
-func TestWatch_Many(t *testing.T) {
+func TestWatchMany(t *testing.T) {
```
Comment (hackerghost93) [inline watcher_test.go:85]: tests with golang v1.22 where timing out. it would never close.

Comment (kelindar) [reply watcher_test.go:85]: What's the reason for this?

Comment (kelindar) [reply watcher_test.go:85]: Let's fix it by disposing in `Close()`:

```go
func (w *watcher) Close() error {
	w.changeState(isRunning, isCanceled)
	w.dispose()
	return nil
}
```

Comment (hackerghost93) [reply watcher_test.go:85]: Fixed !

**Review by kelindar — APPROVED** [review]

**Pull request merged**